docs/rfc/rfc-023-semi-permanent-testnet.md

```diff
@@ -0,0 +1,206 @@
+      subsystems have not deadlocked; e.g. if the watchdog can't acquire and immediately release
+      a particular mutex once in every 5-minute period, it is near certain that the target
+      subsystem has deadlocked, and an alert must be sent to the engineering team.
+      (Outside of the testnet, the watchdogs could be disabled, or they could panic on failure.)
+    - The notification system does some deduplication to minimize spam on system failure
+
+### Testnet long-term vision
+
+The long-term vision includes goals that are not necessary for short- or medium-term success,
+but which would support building an increasingly stable and performant product.
+These goals would generally be beyond the one-year plan,
+and therefore they would not be part of initial planning.
+
+- There is a centralized dashboard to get a quick overview of all testnets,
+  or at least one centralized dashboard per testnet,
+  showing TBD basic information
+- Testnets include cloud spot instances which periodically and abruptly join and leave the network
+- The testnets are a heterogeneous mixture of straight VMs and Docker containers,
+  thereby more closely representing production blockchains
+- Testnets have some manner of continuous profiling,
+  so that we can produce an apples-to-apples comparison of CPU/memory cost of particular operations
+
+### Testnet non-goals
+
+There are some things we are explicitly not trying to achieve with long-lived testnets:
+
+- The Tendermint engineers will NOT be responsible for the testnets' availability
+  outside of working hours; there will not be any kind of on-call schedule
+- As a result of the 8x5 support noted in the previous point,
+  there will be NO guarantee of uptime or availability for any testnet
+- The testnets will NOT be used to gate pull requests;
+  that responsibility belongs to unit tests, end-to-end tests, and integration tests
+- Similarly, the testnet will NOT be used to automate any changes back into Tendermint source code;
+  we will not automatically create a revert commit due to a failed rollout, for instance
+- The testnets are NOT intended to have participation from machines outside of the
+  Tendermint engineering team's control, as the Tendermint engineers are expected
+  to have full access to any instance where they may need to debug an issue
+- While there will certainly be individuals within the Tendermint engineering team
+  who will continue to build out their individual "devops" skills to produce
+  the infrastructure for the testnet, it is NOT a goal that every Tendermint engineer
+  is even _familiar_ with the tech stack involved, whether it is Ansible, Terraform,
+  Kubernetes, etc.
+  As a rule of thumb, all engineers should be able to get shell access on any given instance
+  and should have access to the instance's logs.
+  Little if any further operational skills will be expected.
+- The testnets are not intended to be _created_ for one-off experiments.
```

Comment:
I'm not sure where, but I think we should endeavor to not do any experimentation on these networks. As soon as we introduce config-skew into the networks, we will reduce confidence in the results of running the testnets considerably.

"Wait, was that the node that you changed the max connections on? Maybe it's nothing" etc.

Reply:
f6d33c485 clarifies that it is okay to interact with the blockchain, but configuration should not be modified.